src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -5505,10 +5642,15 @@ char *fd_set_fd_translator_tolind(struct NaClApp* nap, fd_set *fdset, int maxfd,
 void fd_set_fd_translator_fromlind(struct NaClApp* nap, fd_set *fdset, char* otherbitfield, int maxfd) {
   int fds[FD_SETSIZE];
   int fdsindex = 0;
+  struct NaClDesc *ndp;
 
   for(int i = 0; i < maxfd; i++) {
     if(FD_ISSET(i, fdset)) {
-      int translated_fd = fds[fdsindex++] = descnum2Lindfd(nap, i);
+      ndp = GetDescFromCagetable(nap, i);
+      int tempfd = NaClDesc2Lindfd(ndp);
+      int translated_fd = fds[fdsindex++] = tempfd;
+      NaClDescUnref(ndp);
```

Comment:
Would you not need to unref twice as you got the ndp before (in the tolind) without decref'ing it?

Reply:
Yes, I was not sure about that-my thought was since we already ref them once in tolind, we can just unref the same fd-desc in fromlind.

Reply:
Well if I'm not mistaken the refcount is incremented in NaClGetDesc within GetDescFromCagetable which we call twice.